src/Nethermind/Nethermind.Core/Threading/McsLock.cs

```diff
@@ -76,16 +81,26 @@ private static void WaitForUnlock(ThreadNode node, ThreadNode predecessor)
         // the next thread in line may be sleeping when lock is released.
         while (node.State != (nuint)LockState.ReadyToAcquire)
         {
-            if (sw.NextSpinWillYield)
+            if (predecessor.State == (nuint)LockState.Waiting &&
```

Comment:
So this weakens the condition to enter the `WaitForSignal` based on `Monitor` from _next will yield_ to _next will yield AND the predecessor is still waiting_.

Reply:
Yes; so if prior has the lock it will keep spinning but will only be one thread burning cpu (next in line)